src/gui/Src/Gui/MemoryMapView.cpp

```diff
@@ -195,6 +195,37 @@ void MemoryMapView::setupContextMenu()
     connect(Config(), SIGNAL(shortcutsUpdated()), this, SLOT(refreshShortcutsSlot()));
 }
 
+void MemoryMapView::sortRows(int column, bool ascending)

```

Comment:
```cpp
void StdTable::sortRows(int column, bool ascending)
{
    auto sortFn = mColumnSortFunctions.at(column);
    auto addressSortFn = mColumnSortFunctions.at(0);
    std::stable_sort(mData.begin(), mData.end(), [column, ascending, &sortFn](const std::vector<CellData> & a, const std::vector<CellData> & b)
    {
        auto less = sortFn(a.at(column).text, b.at(column).text);
        if(column > 0 && !less)
            less = addressSortFn(a.at(0).text, b.at(0).text));
        return ascending ? less : !less;
    });
}
```

Would something like this work too?

Reply:
No, I had tried something similar to this. The issue arises from:

[https://github.com/x64dbg/x64dbg/blob/b7347f4506e117548b5d32b23a68cd6a5eabba34/src/gui/Src/BasicView/StdTable.cpp#L18](url)
`bool StdTable::SortBy::AsText(const QString & a, const QString & b)
{
    auto i = QString::compare(a, b);
    if(i < 0)
        return true;
    if(i > 0)
        return false;
    return duint(&a) < duint(&b);
}`

Equality collisions are solved (somewhat non-sensibly) by a pointer comparison that will have different results each time the table is sorted. I made a brief attempt to solve the solution here, but the changes I tried always seemed to create issues in other widgets that also use these sort functions.

I very well could be overlooking something simple here, x64dbg's source is quite large and I've only recently begun familiarizing myself with it.

Reply:
Oh that pointer comparison is 100% wrong, likely it was an artifact from before stable_sort was introduced.

Feel free to change that and let me know any issues that arise.

Reply:
Oh lol, that was indeed an easier fix than I thought 😅